jjb/jobs/macros.yaml

```diff
@@ -1,27 +1,13 @@
-# Mark this node offline so that jenkins does not assign jobs to it.  This
-# avoids a race condition created by nodepool: Nodepool receives a notification
-# that a job is finished on a node, and begins to destroy the instance in
-# openstack. Because the node is not marked offline and jenkins is not aware of
-# nodepool, jenkins assigns a job to the node that has been waiting for a node
-# with this label to become available in the short span of time when the single
-# job executor on the nodepool node is not running a job and has not yet been
-# removed from the slave pool by nodepool. The job fails because its slave
-# mysteriously disappears from jenkins. Marking the slave node as offline in a
-# post-build step prevents jenkins from assigning jobs to the node.
+# Delete the slave in order to make it disposable. If another job needs to run
+# on a slave with the same label OpenStack Jenkins plugin will provision a new
+# one using the template as the base image.
 #
-# Marking the slave offline should be the last post-build step on a job, since
-# nodepool also terminates nodes that are marked offline in jenkins. This
-# creates another potential race condition where a slave is deleted before
-# later post-build steps can finish, which also causes the job to fail.
+# Deleting the slave offline must be the last post-build step on a job, since
+# the slave will be set to offline and OpenStack Jenkins plugin will remove it.
+# This will fail the job because, to Jenkins, the slave suddenly went offline.
 - publisher:
```

Comment:
This language is a little unclear. Do you mean the following when you say "This will fail the job..."

```
# If we delete the slave while it is still online, the job will fail, because 
Jenkins will think that the slave suddenly went offline.
```

Reply:
I will clarify. But basically it must be the last thing because the slave running the job will be deleted and then will fail because jenkins won't find it